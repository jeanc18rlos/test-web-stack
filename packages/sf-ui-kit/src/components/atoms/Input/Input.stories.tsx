import Input from './Input';
import { useState } from 'react';
import { StoryWrapper, StoryHeader, StoryHeading } from '../../../utils/stories';
import { Meta } from '@storybook/react';
export default {
  title: 'Components/atoms/Input',
  component: Input,
  argTypes: {
    placeholder: {
      control: { type: 'text' },
      defaultValue: 'Placeholder'
    },
    label: {
      control: { type: 'text' },
      defaultValue: 'Label'
    }
  }
} as Meta;

export const normal = ({ placeholder, label }: any) => {
  const onChange = (e: any) => {
    setValue(e.target.value);
  };
  const [value, setValue] = useState('initialValue');
  return (
    <StoryWrapper>
      <StoryHeader name="Input" />
      <StoryHeading>
        <h2>input</h2>
      </StoryHeading>
      <br />
      <div
        style={{
          width: '100%',
          padding: '20px',
          boxSizing: 'border-box'
        }}
      >
        <Input {...{ value, onChange, placeholder, label }} />
      </div>
    </StoryWrapper>
  );
};

export const search = ({ typeInput, errorMessage, width }: any) => {
  const onChange = (e: any) => {
    setValue(e.target.value);
  };
  const [value, setValue] = useState('initialValue');
  return (
    <StoryWrapper>
      <StoryHeader name="Input" />
      <StoryHeading>
        <h2>input</h2>
      </StoryHeading>
      <br />
      <div
        style={{
          width: '100%',
          padding: '20px',
          boxSizing: 'border-box'
        }}
      >
        <Input {...{ value, onChange, placeholder: 'Search...', typeInput, errorMessage, width }} />
      </div>
    </StoryWrapper>
  );
};
