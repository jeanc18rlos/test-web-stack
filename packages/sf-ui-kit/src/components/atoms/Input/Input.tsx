import React, { memo } from 'react';
import StyledInput from './styled/StyledInput';

interface Props {
  label?: string;
  placeholder: string;
  onChange: any;
  onBlur?: any;
  name?: any;
  inputRef?: Function;
}

const Input = React.forwardRef<HTMLInputElement, Props>(
  ({ onChange, onBlur, name, label, placeholder, ...rest }, ref) => {
    return (
      <StyledInput>
        {label && <label className="title_label">{label}</label>}
        <input
          {...rest}
          name={name}
          ref={ref}
          onChange={onChange}
          onBlur={onBlur}
          type="text"
          placeholder={placeholder}
        />
      </StyledInput>
    );
  }
);

export default memo(Input);
