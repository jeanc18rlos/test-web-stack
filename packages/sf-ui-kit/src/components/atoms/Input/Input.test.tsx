import Input from './Input';

describe('Infobar', () => {
  const onChange = () => {
    alert('Changed');
  };
  it('should render correctly', () => {
    const component = (
      <Input
        label="Label"
        placeholder="Placeholder"
        onChange={onChange}
      ></Input>
    );
    expect(component).toMatchSnapshot();
  });
});
