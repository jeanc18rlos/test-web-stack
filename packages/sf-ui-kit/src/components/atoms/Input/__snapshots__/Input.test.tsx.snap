// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Infobar should render correctly 1`] = `
<Memo
  label="Label"
  onChange={[Function]}
  placeholder="Placeholder"
/>
`;
